rpcs3/Emu/Cell/lv2/sys_net.cpp

```diff
@@ -1595,12 +1610,141 @@ error_code _sys_net_write_dump(ppu_thread& ppu, s32 id, vm::cptr<void> buf, s32
 	return CELL_OK;
 }
 
+error_code lv2_socket::handle_abort(s32 type, u64 arg, s32 flags)
+{
+	enum abort_type : s32
+	{
+		_socket,
+		resolver,
+		type_2, // ??
+		type_3, // ??
+		all,
+	};
+
+	switch (type)
+	{
+	case _socket:
+	{
+		std::map<std::shared_ptr<ppu_thread>, bool> to_awake;
+
+		std::lock_guard nw_lock(g_fxo->get<network_context>().s_nw_mutex);
+
+		const auto sock = idm::get<lv2_socket>(static_cast<u32>(arg));
+
+		if (!sock)
+		{
+			return -SYS_NET_EBADF;
+		}
+
+		{
+			std::lock_guard lock(sock->mutex);
+
+			if (sock->queue.empty())
+			{
+				if (flags & 1)
+				{
+					// Strict error checking: ENOENT if nothing happened
+					return -SYS_NET_ENOENT;
+				}
+
+				// TODO: Abort the subsequent function called on this socket (need to investigate correct behaviour)
+				return CELL_OK;
+			}
+
+			for (auto it = sock->queue.begin(); it != sock->queue.end(); it = sock->queue.erase(it))
+			{
+				if (it->first) to_awake.try_emplace(it->first, true);
+			}
+
+			sock->events.store({});
+		}
+
+		// Check if the PPU reappears in other sockets in case it waits on multiple sockets (so it won't be woken up) 
+		idm::select<lv2_socket>([&](u32, lv2_socket& _sock)
```

Comment:
Is this from hardware tests?
I would expect poll/select to be woken up by an abort on one of their sockets(I might be wrong though).